src/components/calcite-label/calcite-label.tsx

```diff
@@ -58,64 +58,41 @@ export class CalciteLabel {
 
   //--------------------------------------------------------------------------
   //
-  //  Event Listeners
+  //  Variables
   //
   //--------------------------------------------------------------------------
 
-  @Listen("click")
-  onClick(event: MouseEvent): void {
-    const target = event.target as HTMLElement;
-    this.calciteLabelFocus.emit({
-      labelEl: this.el,
-      interactedEl: target,
-      requestedInput: this.for
-    });
-    this.handleCalciteHtmlForClicks(target);
-  }
+  mutationObserver = new MutationObserver(() => this.setEffectiveForElement());
+
+  effectiveForElement: HTMLElement;
 
   //--------------------------------------------------------------------------
   //
   //  Private Methods
   //
   //--------------------------------------------------------------------------
 
-  private handleCalciteHtmlForClicks = (target: HTMLElement) => {
-    // 1. has htmlFor
-    if (!this.for) {
-      return;
-    }
+  labelClickHandler = (event: MouseEvent): void => {
+    const { effectiveForElement } = this;
 
-    // 2. htmlFor matches a calcite component
-    const inputForThisLabel: HTMLElement = queryElementRoots(this.el, `#${this.for}`);
-    if (!inputForThisLabel) {
-      return;
-    }
-    if (!inputForThisLabel.localName.startsWith("calcite")) {
+    if (!effectiveForElement || event.composedPath().includes(effectiveForElement)) {
       return;
     }
 
-    // 5. target is NOT the calcite component that this label matches
-    if (target === inputForThisLabel) {
-      return;
-    }
+    this.calciteLabelFocus.emit({
+      labelEl: this.el,
+      requestedInput: this.for
+    });
 
-    // 3. target is not a labelable native form element
-    const labelableNativeElements = [
-      "button",
-      "input",
-      "meter",
-      "output",
-      "progress",
-      "select",
-      "textarea"
-    ];
-    if (labelableNativeElements.includes(target.localName)) {
-      return;
-    }
+    focusElement(effectiveForElement);
+    effectiveForElement?.click();
+  };
 
-    // 4. target is not a labelable calcite form element
-    const labelableCalciteElements = [
-      "calcite-button",
+  setEffectiveForElement = (): void => {
+    const labelableElements = [
+      // todo: should this list be managed outside? do we need to add any?
```

Comment:
Good question. I'd like to have a pattern where each component implements their label-click behavior instead of having label handle all possible cases. This could also allow external components to be compatible with our label.